app/src/main/java/vn/ztech/software/projectgutenberg/data/repository/source/BookDataSource.kt

```diff
@@ -0,0 +1,15 @@
+package vn.ztech.software.projectgutenberg.data.repository.source
+
+import vn.ztech.software.projectgutenberg.data.model.Book
+import vn.ztech.software.projectgutenberg.data.repository.OnResultListener
+
+interface BookDataSource {
+
+    interface Local {
+        fun getBooksLocal(listener: OnResultListener<MutableList<Book>>)
```

Comment:
```suggestion
        fun getBooksLocal(listener: OnResultListener<List<Book>>)
```
 data đổ ra view sẽ là immutable nha a , nếu vào các case đặc biệt cần modify thì mình sẽ dùng 1 số phương thức mapper